### PR TITLE
HPCC-21779 Std.File.Copy does not have nosplit option

### PR DESCRIPTION
**Pull request by AttilaVamos**
Add nosplit option to:
- Copy
- SprayFixed
- SprayVariable
- SprayXml
- RemotePull
functions.

Handle new parameter in fileservices.

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [x] Send notifications about my Pull Request position in Smoketest queue.
- [x] Test my draft Pull Request.

## Testing:

Tested manually.

### CODE REVIEW

**Comment by AttilaVamos** [comment]
@jakesmith please review.

File: plugins/fileservices/fileservices.cpp [hunk 1]
```diff
@@ -2054,25 +2106,8 @@ FILESERVICES_API void FILESERVICES_CALL fsRemotePull(ICodeContext *ctx,
...
-                                                     int maxConnections,
-                                                     bool overwrite,
-                                                     bool replicate,
-                                                     bool asSuperfile,
-                                                     bool forcePush,
-                                                     int transferBufferSize,
-                                                     bool wrap,
-                                                     bool compress)
+                                                     bool compress,
+                                                     bool noSplit)
```
Comment (jakesmith) [inline plugins/fileservices/fileservices.cpp]: should this have expiryDays too?

File: testing/regress/ecl/split_test.ecl [hunk 2]
```diff
@@ -0,0 +1,574 @@
...
+END;
+
+sprayPrepFileName := prefix + 'spray_prep';
+desprayOutFileName := dropzonePath + WORKUNIT + '-spray_input';
+//sprayOutFileName := prefix + 'spray_test';
+
+allPeople := DATASET([ {'foo', 10, 1},
+                       {'bar', 12, 0},
+                       {'baz', 32, 1}]
+            ,Layout_Person);
```
Comment (jakesmith) [inline testing/regress/ecl/split_test.ecl]: Would be better test if initial set was larger and distributed.
Could do something like:
```
allPeople := DATASET(10000, TRANSFORM(layout_Person, SELF.name := (string)COUNTER; SELF.age := COUNTER%99; SELF.good := IF(COUNTER%2=1, TRUE, FALSE)), DISTRIBUTED);
```

**Comment by jakesmith** [comment]
@AttilaVamos - looks good, a couple of minor comments.

File: plugins/fileservices/fileservices.cpp (hunk 1 above)
Comment (AttilaVamos) [reply plugins/fileservices/fileservices.cpp]: Yes

File: testing/regress/ecl/split_test.ecl (hunk 2 above)
Comment (AttilaVamos) [reply testing/regress/ecl/split_test.ecl]: Yes, but I should test it with SprayFixed().

Comment (AttilaVamos) [reply testing/regress/ecl/split_test.ecl]: Changed and updated for SprayFix().

**Comment by AttilaVamos** [comment]
@jakesmith  Updated, please re-review.

**Comment by jakesmith** [comment]
@AttilaVamos - looks good, please squash.
Should this be 7.6?

**Comment by AttilaVamos** [comment]
@richardkchapman Should this be 7.6.x?

**Comment by AttilaVamos** [comment]
@jakesmith squashed.

**Comment by richardkchapman** [comment]
@AttilaVamos Yes, I would say 7.6 feels more appropriate

**Comment by AttilaVamos** [comment]
@jakesmith Close to re-target. The new PR is: https://github.com/hpcc-systems/HPCC-Platform/pull/12928

**Pull request closed without merging**